### DISCUSSION TITLE
Rending of road names vs ref=*

### DISCUSSION THREAD

**Discussion by repairfox** [osmandapp/OsmAnd#17615] [Q&A]
Hello, I use Osmand for routing. I work on a 911 ambulance. Our addresses are given with the road name. I like to see the road name AND ref on the map. It seems most map styles prefer a ref=* if it exists, and don’t display the name. Offroad and Topo map styles help, but where there are short named ways, it won’t render the name at all, even if the way doesn’t have a ref. In a hamlet where all roads have ref and name tags, it seems to be random it will always show the ref, and sometimes the name, sometimes no.

It seems that whenever the map is rotated such that a road is horizontal (or very near horizontal), the road name will display. As soon as map is rotated while keeping same zoom level the name rendering goes away. I'm in a rural area, so I like my map magnification set at 75% or 50%, I've tried at varias zoom levels with no change.

Tl;dr: make some way for me to choose if I want to render route ref, name, or both.

**Comment by vshcherb** [comment]
Ref and route name have slightly different rendering rules and they are displayed mostly independent .We have quite many fixes to improve name rendering in 4.5.5 release which currently is in beta comparing to 4.4 releases. So it should improve situation soon.

**Reply by Davileci** [reply]
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/97534508/991eaf7e-bad8-4c65-b293-c9d3976123ba"  width="300" height="auto"/> </td>
  </tr>
</table>

[A-625](https://www.openstreetmap.org/way/954724711) and [BI-625](https://www.openstreetmap.org/way/678379147) have the same tags:
- ref:colour = orange
- ref:colour_bg = orange
- ref:colour_tx =	black

An rectangle is rendered for the first, but not for the second. Maybe an excessive number of characters, more than five, breaks the corners as can be seen on E 70 and E 05 versus GI-2132 and GI-2134.

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/97534508/8493ce91-4363-4bb7-9085-9a1be50282cc"  width="300" height="auto"/> </td>
  </tr>
</table>
